x-pack/plugins/maps/public/embeddable/map_embeddable_factory.js

```diff
@@ -28,8 +31,20 @@ export class MapEmbeddableFactory extends EmbeddableFactory {
     this._savedObjectLoader = gisMapSavedObjectLoader;
   }
 
-  async _getIndexPatterns(indexPatternIds = []) {
-    const promises = indexPatternIds.map(async (indexPatternId) => {
+  async _getIndexPatterns(layerListJSON) {
+    // Need to extract layerList from store to get queryable index pattern ids
+    const store = createMapStore();
+    try {
+      JSON.parse(layerListJSON).forEach(layerDescriptor => {
+        store.dispatch(addLayerWithoutDataSync(layerDescriptor));
+      });
+    } catch (error) {
+      // Unable to parse layerListJSON
```

Comment:
I'd prefer throwing and have the caller handle the error.

in any case, I think we should return out of the function, since the ability to parse that json is a prerequisite for the rest no (?).